sdk/src/asset_handlers/woff_io.rs

```diff
@@ -0,0 +1,1739 @@
+// Copyright 2023 Monotype. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+use std::{
+    collections::{
+        BTreeMap,
+    },
+    convert::TryFrom,
+    fs::File,
+    io::{BufReader, Cursor, Read, Seek, SeekFrom, Write},
+    mem::{size_of},
+    path::*,
+};
+
+use byteorder::{BigEndian, ReadBytesExt, WriteBytesExt};
+use log::trace;
+use serde_bytes::ByteBuf;
+use tempfile::TempDir;
+use uuid::Uuid;
+
+use crate::{
+    assertions::BoxMap,
+    asset_io::{
+        AssetBoxHash, AssetIO, CAIRead, CAIReadWrite, CAIReader, CAIWriter, HashBlockObjectType,
+        HashObjectPositions, RemoteRefEmbed, RemoteRefEmbedType,
+    },
+    error::{Error, Result},
+};
+
+/// Debug hook for watching Result<T, Error> values fly past.
+/// 
+/// Debug the code
+///    let my_thing = mysteriously_failing_function(args);
+/// by writing
+///    let my_thing = debug_cough(mysteriously_failing_function(args));
+/// 
+#[allow(dead_code)]
+fn debug_cough<T>(result: Result<T>) -> Result<T> {
```

Comment:
I believe we could get rid of this if we are not using it. If we do want to keep it, I believe it needs to change to:

```rust
fn debug_cough<T, E: std::fmt::Debug>(result: std::result::Result<T, E>) -> std::result::Result<T, E> {
    match result.as_ref() {
        Err(ref e) => {
            trace!("cough: ERROR! {:?}", e);
        },
        Ok(_) => {
            trace!("cough: ok");
        },
    }
    result
}
```

Reply:
Nah, I'm saving it... I need a Rust P/D story I can put this in...

Reply:
OK, would it make sense to update it with what I posted? Otherwise, you can't really call the method unless it is on a `std::result::Result<T, crate::error::Error>`?

Reply:
Notice my posted one has 3 changes uses the `std::result::Result`, specifies `E: fmt::Debug` and then matches on the reference to not take ownership to allow you to return it at the end.